pkg/infraConfig/config/infraConfigClient_ent.go

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import (
+	"fmt"
+	v1 "github.com/devtron-labs/devtron/pkg/infraConfig/bean/v1"
+	"github.com/devtron-labs/devtron/pkg/infraConfig/repository"
+	"github.com/devtron-labs/devtron/pkg/infraConfig/units"
+	"github.com/devtron-labs/devtron/pkg/infraConfig/util"
+	"github.com/go-pg/pg"
+)
+
+type InfraConfigEntClient interface{}
+
+type configFactories struct {
+	cpuConfigFactory     configFactory[float64]
+	memConfigFactory     configFactory[float64]
+	timeoutConfigFactory configFactory[float64]
+}
+
+type unitFactories struct {
+	cpuUnitFactory  units.UnitService[float64]
+	memUnitFactory  units.UnitService[float64]
+	timeUnitFactory units.UnitService[float64]
+}
+
+func (impl *InfraConfigClientImpl) getEntConfigurationUnits() (map[v1.ConfigKeyStr]map[string]v1.Unit, error) {
+	return make(map[v1.ConfigKeyStr]map[string]v1.Unit), nil
+}
+
+func (impl *InfraConfigClientImpl) formatTypedValueAsString(configKey v1.ConfigKeyStr, configValue any) (string, error) {
+	switch configKey {
+	case v1.CPU_LIMIT, v1.CPU_REQUEST:
+		return impl.getCPUConfigFactory().formatTypedValueAsString(configValue)
+	case v1.MEMORY_LIMIT, v1.MEMORY_REQUEST:
+		return impl.getMemoryConfigFactory().formatTypedValueAsString(configValue)
+	case v1.TIME_OUT:
+		return impl.getTimeoutConfigFactory().formatTypedValueAsString(configValue)
+	default:
+		return "", fmt.Errorf("config key %q not supported", configKey)
+	}
```

Comment:
ideally ent files should not contain any business logic. If this are to be used in oss let's move these to normal file

Reply:
use override logic for ent